src/main/java/cubicchunks/server/PlayerCubeMap.java

```diff
@@ -798,4 +805,50 @@ boolean cubePosChanged() {
     public Iterator<CubeWatcher> getRandomWrappedCubeWatcherIterator(int seed) {
         return this.cubeWatchers.randomWrappedIterator(seed);
     }
+    
+    public Iterator<Cube> getCubeIterator() {
+        WorldServer world = this.getWorldServer();
+        final Iterator<CubeWatcher> iterator = this.cubeWatchers.iterator();
+        ImmutableSetMultimap<ChunkPos, Ticket> persistentChunksFor = ForgeChunkManager.getPersistentChunksFor(world);
+        world.profiler.startSection("forcedChunkLoading");
+        final Iterator<Cube> persistentCubesIterator = persistentChunksFor.keys().stream()
+                .filter(Objects::nonNull)
+                .map(input -> ((IColumn)world.getChunkFromChunkCoords(input.x, input.z)).getLoadedCubes())
+                .collect(ArrayList<Cube>::new, (list,cubeCollection)->((ArrayList<Cube>)list).addAll(cubeCollection), (list,cubeList)->((ArrayList<Cube>)list).addAll(cubeList))
```

Comment:
is the ArrayList cast necessary?

Reply:
Yes. Due to code limitation of `collect(...)` `list` is a plain object and contain no methods. I could cast to `List` though.